pytorch_lightning/loops/utilities.py

```diff
@@ -62,6 +64,20 @@ def _extract_hiddens(training_step_output: STEP_OUTPUT, truncated_bptt_steps: in
     return hiddens
 
 
+# rank_zero_warning of type UserWarning for max_epochs if not set
+def _parse_max_epochs_and_steps(
+    max_steps: int, max_epochs: int, max_time: Union[str, timedelta, Dict[str, int]], min_steps: int, min_epochs: int
+) -> Tuple[int, int]:
+    if max_epochs is None and max_steps == -1 and max_time is None:
```

Comment:
We should rewrite it to this structure:

```
if condition:
    emit warning that max_epochs=1000
    set max_epochs=1000
```
This way it is clearer under which condition the warning applies and what the end value will be.

Reply:
If I understood correctly, then it should be the following

```
if max_epochs is None:
     rank_zero_warn(...)
     min_epochs = (1 if (min_epochs is None and min_steps is None and max_time is None) else min_epochs)
     max_epochs = max_epochs if max_epochs is not None else (1000 if (max_steps == -1 and max_time is None) 
        else -1)
    return min_epochs, max_epochs
```
Is this correct? Thanks